pkg/agent/openflow/pipeline.go

```diff
@@ -454,388 +504,472 @@ func (c *client) DeleteOFEntries(ofEntries []binding.OFEntry) error {
 				Done(),
-			ConntrackCommitTable.BuildFlow(priorityLow).MatchProtocol(proto).
-				MatchCTStateNew(true).MatchCTStateTrk(true).
-				Action().CT(true, ConntrackCommitTable.GetNext(), ctZone).
+			// This generates the flow to match the first packet of non-Service connection and mark the source of the connection
+			// by copying PktSourceField to ConnSourceCTMarkField.
+			ConntrackCommitTable.ofTable.BuildFlow(priorityNormal).
+				Cookie(cookieID).
+				MatchProtocol(ipProtocol).
+				MatchCTStateNew(true).
+				MatchCTStateTrk(true).
+				Action().CT(true, ConntrackCommitTable.GetNext(), f.ctZones[ipProtocol]).
 				MoveToCtMarkField(PktSourceField, ConnSourceCTMarkField).
 				CTDone().
-				Cookie(c.cookieAllocator.Request(category).Raw()).
 				Done(),
 		)
 	}
+	// This generates default flow to match the first packet of a new connection and forward it to stagePreRouting.
+	flows = append(flows, ConntrackStateTable.ofTable.BuildFlow(priorityMiss).
+		Cookie(cookieID).
+		Action().GotoStage(stagePreRouting).
+		Done())
+
 	return flows
 }
 
-func (c *client) kubeProxyFlows(category cookie.Category) []binding.Flow {
+// Feature: featureService
+// Stage: stageConntrackState
+// Tables: ConntrackStateTable, ConntrackTable
+// Refactored from:
+//   - part of func (c *client) connectionTrackFlows(category cookie.Category) []binding.Flow
+//   - part of func (c *client) serviceLBBypassFlows(ipProtocol binding.Protocol) []binding.Flow
+// conntrackFlows generates the flows about conntrack for feature Service.
+func (f *featureService) conntrackFlows() []binding.Flow {
+	cookieID := f.cookieAllocator.Request(f.category).Raw()
 	var flows []binding.Flow
-	for _, proto := range c.ipProtocols {
-		ctZone := CtZone
-		if proto == binding.ProtocolIPv6 {
-			ctZone = CtZoneV6
-		}
+	for _, ipProtocol := range f.ipProtocols {
 		flows = append(flows,
-			ConntrackTable.BuildFlow(priorityNormal).MatchProtocol(proto).
-				Action().CT(false, ConntrackTable.GetNext(), ctZone).CTDone().
-				Cookie(c.cookieAllocator.Request(category).Raw()).
+			// This generates the flow to mark tracked DNATed Service connection with RewriteMACRegMark (load-balanced by
+			// AntreaProxy) and forward the packets to stageEgressSecurity directly to bypass stagePreRouting.
+			ConntrackStateTable.ofTable.BuildFlow(priorityNormal).
+				Cookie(cookieID).
+				MatchProtocol(ipProtocol).
+				MatchCTMark(ServiceCTMark).
+				MatchCTStateNew(false).
+				MatchCTStateTrk(true).
+				Action().LoadRegMark(RewriteMACRegMark).
+				Action().GotoStage(stageEgressSecurity).
+				Done(),
+			// This generates the flow to skip ConntrackCommitTable for Service connection (with ServiceCTMark). Since
+			// Service connection has been committed in EndpointDNATTable in DNAT CT zone, it's unnecessary to commit
+			// Service connection in ConntrackCommitTable again since commit operation in EndpointDNATTable and
+			// ConntrackCommitTable use the same CT zone.
```

Comment:
This generates the flow to avoid committing Service connections (with ServiceCTMark) another time. They have been committed in EndpointDNATTable, using the same CT zone.

Reply:
Done.